### DISCUSSION TITLE
Error: "Passwords cannot contain re-hashed personal information."

### DISCUSSION THREAD

**Discussion by MarcPinnell** [codeigniter4/shield#482] [Q&A] [answered]
Not sure if this is a bug or my lack of understanding (since I am new to Shield). I am getting that error on submitting a new registration (custom controller) with the following information (field name >> value):

```
fname >> Steve
lname >> Perry
phone_home >> 5305551212
email >> steve@journey.com
password >> v5ZW3V^4^R!^
password_confirm >> v5ZW3V^4^R!^
```

When I submit that it fails with the mentioned error. If I remove validation, it saves fine.

Controller (work in progress, do it one step at a time to try and understand):

```php
	public function register_insert()
	{
		if (auth()->loggedIn()) {
			return redirect()->to(config('Auth')->registerRedirect());
		}

		$post = $this->request->getPost();

		$post['phone_home'] = cleanPhone($post['phone_home']);

		$users = model('Userp42Model');

		// Validate here first, since some things,
		// like the password, can only be validated properly here.
		$rules = $this->getValidationRules();

		if (! $this->validate($rules)) {
			return redirect()->back()->withInput()->with('errors', $this->validator->getErrors());
		}

		$user = new User();
		$user->fill($post);
		$users->save($user);

		// We need an ID to on the entity to save groups.
		if ($user->id === null) {
			$user->id = $users->getInsertID();
		}

		// Save the user's groups
		$user->syncGroups(...$this->request->getPost('groups'));

		auth()->logout($user->id);

		return view('Public/registration_thankyou');
	}

```

What in that password is a rehash of personal information?

**Comment by MarcPinnell** [comment]
A little tidbit of additional info. I tried the password "Thisissecret!" and it completed the registration successfully.

**Comment by kenjis** [comment]
The error comes from **NothingPersonalValidator**.
See https://github.com/codeigniter4/shield/blob/develop/docs/concepts.md#password-validators

**Comment by MarcPinnell** [comment]
I know where it is coming from, my question was more along the lines of "Why?". Why is that password of random letters/symbols failing with the input I showed. There is NO similarity of data.

Seems like the first random password is way better than that second one that is just a couple of strung together words.

**Reply by kenjis** [reply]
Probably this: #425

**Comment by kenjis** [comment, marked as answer]
Are you using released version of Shield?
If so, it has many bugs. Please update to `develop`.

**Reply by MarcPinnell** [reply]
Yes, using the released version of Shield. I don't see any version numbers, but I installed it about 2 weeks ago.

Assuming that using "develop" means I get bug fixes sooner?

**Reply by kenjis** [reply]
Yes, the bug has been fixed in `develop` branch.

**Comment by MarcPinnell** [comment]
Forgive me for one last question, but I am new to composer... How do I change branches at this point? This line in the composer.json file?

` "codeigniter4/shield": "^1.0@beta",`

**Reply by kenjis** [reply]
No problem.
`"codeigniter4/shield": "dev-develop"`

See also https://github.com/codeigniter4/shield/blob/develop/docs/install.md#important-composer-error
